sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -609,6 +609,9 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
   require(hset != null, "hset could not be null")
 
   override def toString: String = {
+    if (!child.resolved) {
+      return s"$child INSET (values with unresolved data types)"
```

Comment:
Can we match the string representation? At least I can't find anything by `git grep -r "values with unresolved"`

Reply:
Thanks for raising that. I had this concern too, but I'm not sure which descriptions would be suitable here. I didn't see any similar cases in other toString representations.

Reply:
shall we just use `hset.toSeq.mkString(", ")`?

Reply:
Will there be values can't be converted using mkString? Because there are no type constraints, hset is Seq[Any].

Reply:
All Java objects have `toString`. It may be different from the pretty string defined by Spark SQL, but better than nothing?